internal/generate/olm-catalog/csv_go_test.go

```diff
@@ -472,3 +528,52 @@ func TestUpdateCSVVersion(t *testing.T) {
 		t.Errorf("Wanted csv replaces %s, got %s", wantedReplaces, csv.Spec.Replaces)
 	}
 }
+
+func getCSVFromFile(t *testing.T, path string) *olmapiv1alpha1.ClusterServiceVersion {
+	csvpath := filepath.Join(path)
+	b, err := ioutil.ReadFile(csvpath)
+	if err != nil {
+		t.Errorf("error reading manifest %s: %v", path, err)
+	}
+
+	csv, err := generateCSV(b)
+	if err != nil {
+		t.Errorf("error generating csv %s: %v", path, err)
+	}
+	return csv
+}
+
+func generateCSV(input []byte) (*olmapiv1alpha1.ClusterServiceVersion, error) {
+	scanner := k8sutil.NewYAMLScanner(bytes.NewBuffer(input))
+	for scanner.Scan() {
+		manifest := scanner.Bytes()
+		typeMeta, err := k8sutil.GetTypeMetaFromBytes(manifest)
+		if err != nil {
+			log.Debugf("Skipping non-Object manifest : %v", err)
+			continue
+		}
+		if typeMeta.Kind == olmapiv1alpha1.ClusterServiceVersionKind {
+			csv := &olmapiv1alpha1.ClusterServiceVersion{}
+			if err := yaml.Unmarshal(manifest, csv); err != nil {
+				return nil, fmt.Errorf("error unmarshalling ClusterServiceVersion from manifest: %v", err)
+			}
+			return csv, nil
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("error scanning manifest  %v", err)
+	}
+	return nil, fmt.Errorf("error getting manifest")
+
+}
+
+// removeSDKstampsFromCSV removes the sdk stamps from CSV struct. Used for test cases where
+// we need to test the generated CSV with expected predefined CSV file on disk.
+func removeSDKstampsFromCSVHelper(csv *v1alpha1.ClusterServiceVersion) {
```

Comment:
This could be done more easily for tests by regexp replacing annotations on CSV bytes. Then you don't have to re-marshal CSV's in each test.

Reply:
I had tried using regex to do this. But manipulating yaml in string format and making a string comparison again seems error prone due to spaces.

Reply:
On second thought this is fine for now. Legacy code will be removed soon anyway.